### PR TITLE
Fix: use cloud credentials to look up random regions

### PR DESCRIPTION
**Pull request by jeefy**
If a region is set to random, attempt to look up the opted-in regions available for the cloud provider.

If the credentials aren't set, default to a sane/typical region. 

If not random, fall through.

### CODE REVIEW

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Comment by jeefy** [comment]
/hold
for @whereswaldon

File: pkg/common/providers/ocmprovider/cluster.go [hunk 1]
```diff
@@ -204,15 +204,35 @@ func (o *OCMProvider) DetermineRegion(cloudProvider string) (string, error) {
 	// If a region is set to "random", it will poll OCM for all the regions available
 	// It then will pull a random entry from the list of regions and set the ID to that
 	if region == "random" {
-		regionsClient := o.conn.ClustersMgmt().V1().CloudProviders().CloudProvider(cloudProvider).Regions().List()
+		var regions []*v1.CloudRegion
+		// We support multiple cloud providers....
+		if cloudProvider == "aws" {
+			if viper.GetString(AWSAccessKey) == "" && viper.GetString(AWSSecretKey) == "" {
```
[the remark this answers is not in the document]
Comment (jharrington22) [reply pkg/common/providers/ocmprovider/cluster.go:210]: Both are required to auth to AWS

[the remark this answers is not in the document]
Comment (jeefy) [reply pkg/common/providers/ocmprovider/cluster.go:210]: Thank you. We need both, I'll update quick.

[the remark this answers is not in the document]
Comment (jeefy) [reply pkg/common/providers/ocmprovider/cluster.go]: Updated.

[the remark this answers is not in the document]
Comment (vkareh) [reply pkg/common/providers/ocmprovider/cluster.go]: Thanks @whereswaldon for catching that - totally escaped me :/

**Comment by vkareh** [comment]
/lgtm

this time i actually checked the logic :rofl:

**Pull request merged**